app/bucketlist/views.py

```diff
@@ -0,0 +1,130 @@
+from app.models import User, BucketList, Item
+from flask import request, jsonify, Blueprint
+from app import db, jwt
+from flask_jwt import jwt_required, current_identity
+from app.auth.validate import validate_json
+from datetime import datetime
+from sqlalchemy.sql import func
+
+
+bucketlists = Blueprint('bucketlists', __name__,
+                        url_prefix='/api/v1/bucketlists')
+
+
+# Get all bucketlists and search bucketlists
+@bucketlists.route('/', methods=['GET'])
+@jwt_required()
+def get_bucketlist():
+    search = request.args.get('q', '')
+    page = request.args.get('page', 1)
+    limit = request.args.get('limit', 20)
+
+    results = BucketList.query.filter_by(user_id=current_identity.id).filter(
+        BucketList.name.ilike(f'%{search}%')).paginate(int(page), int(limit), False)
+    if not results:
+        return jsonify({"error": "BucketList not found"}), 404
+
+    buckets = [bucket.to_json() for bucket in results.items]
+
+    return jsonify(buckets), 200
+
+
+# Add new bucketlist
+@bucketlists.route('/', methods=['POST'])
+@jwt_required()
+@validate_json('name')
+def new_bucketlist():
+    data = request.get_json()
+
+    bucket = BucketList.query.filter_by(name=data['name']).first()
+    if bucket:
+        return jsonify({'error': 'BucketList already exists'}), 409
+
+    bucket = BucketList(name=data['name'], user_id=current_identity.id)
+    db.session.add(bucket)
+    db.session.commit()
+    return jsonify({'message': 'BucketList added successfully'}), 201
+
+
+# Update single bucketlist
+@bucketlists.route('/<int:bucket_id>', methods=['PUT'])
+@jwt_required()
+@validate_json('name')
+def update_bucketlist(bucket_id):
+    data = request.get_json()
+    return any_request(request.method, BucketList, bucket_id, data)
+
+
+# Get and Delete single bucketlist
+@bucketlists.route('/<int:bucket_id>', methods=['GET', 'DELETE'])
+@jwt_required()
+def bucketlist(bucket_id):
+    return any_request(request.method, BucketList, bucket_id, None)
+
+
+# Get all bucketlist items
+@bucketlists.route('/<int:bucket_id>/items/', methods=['GET'])
+@jwt_required()
+def get_bucketlist_item(bucket_id):
+    all_items = [item.to_json() for item in
+                 Item.query.filter_by(bucket_id=bucket_id).all()]
+    return jsonify(all_items), 200
+
+
+# Create new  bucketlist item
+@bucketlists.route('/<int:bucket_id>/items/', methods=['POST'])
+@jwt_required()
+@validate_json('name')
+def new_bucketlist_item(bucket_id):
+    data = request.get_json()
+    item = Item.query.filter_by(name=data['name']).first()
```

Comment:
Allow different users to create bucketlist items with the same name and a single user to have bucketlist items with the same name but under different bucketlists.
`item = Item.query.filter_by(name=data['name']).filter_by(bucket_id=bucket_id).filter_by(bucket.user=current_user.id).first()`